tests/epyccel/modules/arrays.py

```diff
@@ -324,20 +324,20 @@ def array_float_4d_C_array_initialization(x : 'float[:,:,:]', y : 'float[:,:,:]'
 
 def array_float_nested_C_array_initialization(x : 'float[:,:,:]', y : 'float[:,:]', z : 'float[:,:]', a : 'float[:,:,:,:]'):
     from numpy import array
-    tmp      = array([x, [y, z, z], x], dtype='float')
+    tmp      = array((x, (y, z, z), x), dtype='float')
```

Comment:
`[x, [y, z, z], x]` is evaluated as a list. It raises the error:
> ERROR at annotation (semantic) stage
> pyccel:
> |fatal [semantic]: tmp_arr.py [3,21]| Inhomogeneous lists are not supported by Pyccel. Please use a tuple ((x, (y, z, z), x))

This was not raised previously because lists and tuples were not cleanly separated.

Reply:
Ah okey I see.

Reply:
Before we would have treated the argument as an object with:
- dtype : float
- precision : 8
- rank : 4
- order : c
- class_type : list

This object is homogeneous but information will be lost when printing.



After this PR the argument is understood as:
- class_type : list[numpy.ndarray[float], tuple[numpy.ndarray[float], ...], numpy.ndarray[float]]
- rank : 4
- order : c

This object is inhomogeneous